test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -930,6 +930,100 @@ test.describe( 'Navigation block', () => {
 			await expect( linkControl.getSearchInput() ).not.toBeVisible();
 		} );
 	} );
+
+	test.describe( 'Navigation colors', () => {
+		test.beforeEach( async ( { requestUtils } ) => {
+			await Promise.all( [ requestUtils.deleteAllMenus() ] );
+			await requestUtils.createNavigationMenu( {
+				title: 'Test Menu',
+				content:
+					'<!-- wp:navigation-submenu {"label":"First link","type":"custom","url":"http://www.wordpress.org/","kind":"custom"} --><!-- wp:navigation-link {"label":"Second Link","type":"custom","url":"http://www.wordpress.org/","kind":"custom"} /--><!-- /wp:navigation-submenu -->',
+			} );
+		} );
+
+		test( 'As a user I expect my navigation links to have good default colors', async ( {
+			admin,
+			editor,
+			requestUtils,
+		} ) => {
+			//we want emptytheme because it doesn't have any styles
+			await requestUtils.activateTheme( 'emptytheme' );
+
+			await admin.createNewPost();
+			await editor.insertBlock( { name: 'core/navigation' } );
+
+			const firstlink = editor.canvas.locator(
+				`role=textbox[name="Navigation link text"i] >> text="First link"`
+			);
+			//Expect the first link to default to black when the theme doesn't define a link color
+			await expect( firstlink ).toHaveCSS( 'color', 'rgb(0, 0, 0)' );
+			//TODO check frontend on desktop and mobile
+			//TODO then the same for second link (make sure it has a background color!)
+			//Then the same for background colors
+		} );
+
+		test( 'As a user I expect my navigation links to inherit the colors from the theme', async ( {
+			admin,
+			editor,
+		} ) => {
+			//setup: define a link color in the theme
+			await admin.createNewPost();
+
+			await editor.insertBlock( { name: 'core/navigation' } );
+
+			const firstlink = editor.canvas.locator(
+				`role=textbox[name="Navigation link text"i] >> text="First link"`
+			);
+
+			await admin.page.pause();
+			//Expect the first link to inherit the link color from the theme.
+			//In this case TT3 defines the link colors inside the post-content block to be --wp--preset--color--secondary = #345C00
+			await expect( firstlink ).toHaveCSS( 'color', 'rgb(52, 92, 0)' );
```

Comment:
With these colors that rely on theme behaviour are they going to remain stable? Another (more complex) option would be to create our own test theme and have "known" colors in it. We could then activate that and assert on values we know will not change unless they are modified within the Gutenberg repo. It could even be a child theme of emptytheme.

As a minimum we should consider extracting all the CSS colors to well named variables so it's easy to update them if/when the test starts to fail.

Reply:
I thought about this, and I don't think having our own theme to test would be too complicated. I'm not expecting to have tt3 colors changing anytime soon, but a custom theme is something we will have control over, so that makes sense.